tests/components/tuya/test_config_flow.py

```diff
@@ -83,192 +102,98 @@ async def test_user(hass, tuya):
+    """Test smart home user config."""
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=TUYA_USER_DATA
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["errors"] == {"base": RESULT_AUTH_FAILED}
-
-
-async def test_abort_on_connection_error(hass, tuya):
-    """Test when we have a network error."""
-    tuya().init.side_effect = TuyaNetException("Boom")
-
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=TUYA_USER_DATA
-    )
-
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == RESULT_CONN_ERROR
-
+    assert result["step_id"] == "user"
 
-async def test_options_flow(hass):
-    """Test config flow options."""
-    config_entry = MockConfigEntry(
-        domain=DOMAIN,
-        data=TUYA_USER_DATA,
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=TUYA_SMART_HOME_PROJECT_DATA
     )
-    config_entry.add_to_hass(hass)
-
-    # Set up the integration to make sure the config flow module is loaded.
-    assert await hass.config_entries.async_setup(config_entry.entry_id)
-    await hass.async_block_till_done()
-
-    # Unload the integration to prepare for the test.
-    with patch("homeassistant.components.tuya.async_unload_entry", return_value=True):
-        assert await hass.config_entries.async_unload(config_entry.entry_id)
-        await hass.async_block_till_done()
-
-    # Test check for integration not loaded
-    result = await hass.config_entries.options.async_init(config_entry.entry_id)
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == RESULT_CONN_ERROR
-
-    # Load integration and enter options
-    await hass.config_entries.async_setup(config_entry.entry_id)
     await hass.async_block_till_done()
-    hass.data[DOMAIN] = {TUYA_DATA: MockTuya()}
-    result = await hass.config_entries.options.async_init(config_entry.entry_id)
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "init"
-
-    # Test dev not found error
-    result = await hass.config_entries.options.async_configure(
-        result["flow_id"],
-        user_input={CONF_LIST_DEVICES: [f"light-{LIGHT_ID_FAKE1}"]},
-    )
-
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "init"
-    assert result["errors"] == {"base": ERROR_DEV_NOT_FOUND}
-
-    # Test dev type error
-    result = await hass.config_entries.options.async_configure(
-        result["flow_id"],
-        user_input={CONF_LIST_DEVICES: [f"light-{LIGHT_ID_FAKE2}"]},
-    )
+    assert result["step_id"] == "login"
+
+    with patch(
+        "homeassistant.components.tuya.config_flow.TuyaConfigFlow._try_login",
```

Comment:
Please patch the library instead of our code in the config flow. See other test where we set the return value of the tuya fixture.

Reply:
Sorry, I don’t particularly understand what this means

Reply:
See this example:
https://github.com/home-assistant/core/blob/4b986f6c4a2f4e23cd8c766d21e6e1b4f4a3a285/tests/components/tuya/test_config_flow.py#L191